iothub/device/src/InternalClient.cs

```diff
@@ -149,6 +151,23 @@ public InternalClient(IotHubConnectionString iotHubConnectionString, ITransportS
             if (Logging.IsEnabled) Logging.Associate(this, transportSettings, nameof(InternalClient));
             _transportSettings = transportSettings;
 
+            if (options != null)
+            {
+                var fileUploadTransportSettings = options.FileUploadTransportSettings;
```

Comment:
Even though you provide a default impl, they could still set it to null, which would give you an NRE here